### PR TITLE
Update to Python 3, Boto3, Boto3 resources, and Improved Logging

### PR DESCRIPTION
**Pull request by matt0x6F**
* Updated to Python 3
* Using Boto3 with resource declaration for cleaner code
* Using Colorama and defaults to silent output with system logging enabled for better cron implementation

### CODE REVIEW

File: backup.py [hunk 1]
```diff
@@ -1,207 +1,195 @@
...
 from subprocess import call
+from colorama import init
+from termcolor import colored
 
-import click
+init()
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+DEFAULT_REGION='us-west-2'
```
Comment (izakp) [inline backup.py]: I'd go with `'us-east-1'` as `DEFAULT_REGION` as that's what I generally see elsewhere.

File: backup.py [hunk 2]
```diff
@@ -1,207 +1,195 @@
...
-@click.option('--exclude', '-e', type=click.STRING, multiple=True, help='Additional direcoties to exclude from the backup')
-@click.option('--dry-run', type=click.BOOL, is_flag=True, help='Create tar archive as "tmp" but to do not upload to S3  : defaults to false')
-def create(ctx, jenkins_home, tmp, tar, tar_opts, exclude_vcs, ignore_fail, exclude_archive, exclude_target,
-            exclude_builds, exclude_workspace, exclude_maven, exclude_logs, exclude, dry_run):
-  """Create a backup"""
-  print("Backing up %s to %s/%s..." % (jenkins_home, ctx.obj['BUCKET'], ctx.obj['BUCKET_PREFIX']))
-
-  command = [tar, tar_opts, tmp, '-C', jenkins_home]
-
-  if exclude_vcs:
```
Comment (izakp) [inline backup.py:91]: Why drop the exclude options?

**Review by izakp — CHANGES_REQUESTED** [review]
Nicely done!  This was an old script and I have since adopted boto3 elsewhere.  And nice logging is a great addition.  Also good to upgrade to Python3 :)

Just a couple of comments about defaults and options for the backup.

File: backup.py (hunk 2 above)
Comment (matt0x6F) [reply backup.py:91]: I can probably drop those back in, I removed them because we weren't using it on my project.

File: backup.py (hunk 1 above)
Comment (matt0x6F) [reply backup.py]: That sounds agreeable.

**Comment by purplehat7** [comment]
Have you tested the new script? I get:

```
Traceback (most recent call last):
  File "/usr/local/bin/backup-jenkins", line 11, in <module>
    load_entry_point('jenkins-backup-s3==0.1.2', 'console_scripts', 'backup-jenkins')()
  File "/usr/local/lib/python2.7/dist-packages/jenkins_backup_s3/backup.py", line 216, in main
    cli(obj={})
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/jenkins_backup_s3/backup.py", line 109, in create
    if exclude_vcs:
NameError: global name 'exclude_vcs' is not defined
```

**Comment by matt0x6F** [comment]
I missed dropping in the args. I've done some testing on my Jenkins box and everything works. That said, some formal tests should be done in a docker container if this is to be developed on in the future.

**Review by izakp — APPROVED** [review]

File: backup.py [hunk 3]
```diff
@@ -80,128 +110,118 @@ def cli(ctx, bucket, bucket_prefix, bucket_region):
 @click.option('--exclude-maven/--include-maven', default=True, help='Exclude maven repository from the backup : defaults to true')
 @click.option('--exclude-logs/--include-logs', default=True, help='Exclude logs from the backup : defaults to true')
 @click.option('--exclude', '-e', type=click.STRING, multiple=True, help='Additional direcoties to exclude from the backup')
-@click.option('--dry-run', type=click.BOOL, is_flag=True, help='Create tar archive as "tmp" but to do not upload to S3  : defaults to false')
```
Comment (izakp) [inline backup.py:83]: Thanks, nice catch :P

**Pull request merged**